---
title: April 28th Notes
layout: post
date: 2015-04-28
category : notes
tags : []
---
{% include JB/setup %}

### daily notes from April 28th, 2015

1. data time from 9:30 to 11 am
2. 11 am to 12:30 - worked on script to *in silico* test error rate on clustering of large data sets
  * took 1000 16S sequences from greengenes
  * used bwa synthesis module to make amplicon copies with various error (0.01, 0.02, 0.03, etc.) rates

  [Schloss vs. Edgar](http://www.mothur.org/forum/viewtopic.php?f=6&t=2572)

### notes from April 29th, 2015

1. data time from 9:30 to 11 am

2. 11 am into afternoon - worked on script to *in silico* test error rate on clustering of large data sets

Original script idea:
  * took 1000 16S sequences from greengenes in rough composition of normal alpha diversity curve for soil
  * used bwa synthesis module to make amplicon copies with various error (0.01, 0.02, 0.03, etc.) rates
  * Need better consideration for
    1. Taq polymerase data errors, and
    2. overall sequencing error of Illumina platform and read degeneration after (average) of 600 cycles for MiSeq run

Need better integration with actual Mock community sequencing to understand the role of amplicon PCR fidelity and sequencing error on number of OTUs generated from mock community.

## QP dissertation defense notes

Title: Novel computational approaches to assess microbial diversity

alpha vs. beta diversity key ecological measurements

Handelsman 1998 : "metagenome" coined

Jack Gilbert 2013 paper - gram of soil contains 4 petabase pairs of DNA

grand prairie challenge data set example of a ton of data and how to make sense of it?

cartoon of human genome assembly - 3 billion piece puzzle

Problems with metagenomics:
  1. **Assembly** (only small porportion of reads assemble - less than 20%, usually much less)
  2. **Binning** for assembly and other - need information on reference genomes and those we have information for are skewed to human associated
  3. **Mapping** computationally expensive, accuracy is a problem, huge ammounts of data

k-mers provide novel information on sequences - use of sliding window

program is `count-min sketch` and it uses a bloom filter approach

IGS -> "informative genomic segment" (QP's replacement for the concept of OTU in metagenome data)

"most unique k-mers are erroneous"

See Imelfort et al 2014 paper - contig converage profile

IGS = number of reads across all samples per k-mer / read coverage

QP used test case data from
  1. Sorcerer I & II marine metagenome sequencing (Ventner projects)
  2. Amazon Rain Forest Microbial Diversity Observatory, see [blog post](http://ivory.idyll.org/blog/jgr-assembling-the-amazon.html) (ARMO data set; Jim Tiedje)
  3. Grand Prairie Challenge Data (Jim Tiedje)

Use of median k-mer coverage (key aspect of comparing "individuals" between samples)

QP is able to use IGS to measure alpha and beta diversity between samples without using BLAST or assembly.

#### Josh's Questions

1. How does IGS compare to OTUs from 16S -- do the techniques compare directly?
2. We want to know taxonomy and function -- can you use the information from IGS to bin sequence reads to better assemble metagenomes to get taxonomy and function?
3. strain level assessment -- how does this k-mer approach deal with microbial strains?
4. Compute IGS before or after diginorm?

____________________________________

Estimated 50 terabytes of unique data Jason Gams 2005

4 petabase pairs of DNA Gilbert 2013

Nature reviews genetics 805-814

Chao estimator

dbg-graph-null for generating random reads

## Notes from April 30th, 2015

9:30 am to 11:00 am - Continued working on sequencing error script from day before.

11:00 am to 12:00 - Xuemei Chen presentation

12:00 to 5:00 pm - figuring out baseline error rates and how that would translate to OTU


#### Notes on sequencing errors and proliferation of OTUs for today

See following papers:

  1. [Insight into biases and sequencing errors for amplicon sequencing with the Illumina MiSeq platform](http://nar.oxfordjournals.org/content/early/2015/01/13/nar.gku1341.full)

  2. [Characterizing and measuring bias in sequence data](http://link.springer.com/article/10.1186/gb-2013-14-5-r51)

  3. [A tale of three next generation sequencing platforms: comparison of Ion Torrent, Pacific Biosciences and Illumina MiSeq sequencers](http://www.biomedcentral.com/1471-2164/13/341)

[Other post on MiSeq errors on 600 cycle run](http://seqanswers.com/forums/showthread.php?t=40879)

#### Error rates of polymerase from experimental data

[Error Rate Comparison during Polymerase Chain Reaction by DNA Polymerase](http://www.hindawi.com/journals/mbi/2014/287430/)

[New England Biolabs DNA Polymerase Selection Chart](https://www.neb.com/tools-and-resources/selection-charts/dna-polymerase-selection-chart)

**Table: Error rates of different brands of polymerase from above paper**

| Enzyme | Average doubling per reaction (192 rxns)  | Error rate (percent) | Code |
|--------------|---------------------------|------------------------------|----------------|
| Crimson & Taq Standard DNA Polymerase | NA | 0.000285 | Crimson |
| VentR® (exo–) DNA Polymerase | NA | 0.00019 | VentRexo |
| Large (Klenow) Fragment | NA | 0.00018 | LargeKlenow |
| LongAmp® & OneTaq® Polymerase | NA | 0.00014 | LongAmp |
| Klenow Fragment (3′→5′ exo-) | NA | 0.0001 | Klenow |
| VentR® DNA Polymerase | NA | 0.000057 | VentR |
| Taq | 18.6 +/- 0.95 | 0.000043 | Taq |
| T7 DNA Polymerase (unmodified) | NA | 0.000015 | T7 |
| AccuPrime-Taq  | 16.95 +/- 0.90 | 0.00001  | AccuPrime |
| E. coli DNA Polymerase I | NA | 0.000009 | Ecoli |
| KOD | 19.2 +/- 1.15 | 0.0000076  | KOD |
| Pfu (cloned) | 14.25 +/- 1.00 | 0.0000028  | Pfu |
| Phusion | 18.8 +/- 1.50 | 0.0000026  | Phusion |
| Pwo | 20.05 +/- 0.90 | 0.0000024  | Pwo |
| Template Lesions T4 DNA Polymerase | NA | 0.000001 | TemplateT4 |
| Flex & Phusion® DNA Polymerase | NA | 0.00000044 | Flex |

[Earth Microbiome protocol](http://www.earthmicrobiome.org/emp-standard-protocols/16s/) reports pcr for 35 cycles.

#### script desciption notes

Wrote script for each error polymerase type - see: ```mock_OTU_qsub_write.sh``` for qsub script generator.

[wgsim](https://github.com/lh3/wgsim) won't work for amplicon data, so I built off of the Grinder script from this paper: [Grinder: a versatile amplicon and shotgun sequence simulator](http://nar.oxfordjournals.org/content/40/12/e94)

For Grinder to work, the script needs to be compiled with perl on linux based computers (BSD Mac); ```cpan``` needs to be invoked (for my perl installation on Mac, YMMV on other linux kernels):
```
sudo cpan -i Getopt::Euclid && sudo cpan -i Math::Random::MT
```

## Notes from May 1st, 2015

### notes (duplicated) for mock community

Deinococcus radiodurans - 3 copies
Burkholderia th. - 4 copies
Bacillus cereus - 13 copies
Pseudomonas syringae - 5 copies
Flavobacterium johnsoniae - 6 copies
Escherichia coli MG1655 - 8 copies
