---
title: QP dissertation notes
layout: post
date: 2015-04-29
category : notes
tags : [microbial diversity, alpha diversity, beta diversity]
---
{% include JB/setup %}

## QP dissertation defense notes

Title: Novel computational approaches to assess microbial diversity

alpha vs. beta diversity key ecological measurements

Handelsman 1998 : "metagenome" coined

Jack Gilbert 2013 paper - gram of soil contains 4 petabase pairs of DNA

grand prairie challenge data set example of a ton of data and how to make sense of it?

cartoon of human genome assembly - 3 billion piece puzzle

Problems with metagenomics:
  1. **Assembly** (only small porportion of reads assemble - less than 20%, usually much less)
  2. **Binning** for assembly and other - need information on reference genomes and those we have information for are skewed to human associated
  3. **Mapping** computationally expensive, accuracy is a problem, huge ammounts of data

k-mers provide novel information on sequences - use of sliding window

program is `count-min sketch` and it uses a bloom filter approach

IGS -> "informative genomic segment" (QP's replacement for the concept of OTU in metagenome data)

"most unique k-mers are erroneous"

See Imelfort et al 2014 paper - contig converage profile

IGS = number of reads across all samples per k-mer / read coverage

QP used test case data from
  1. Sorcerer I & II marine metagenome sequencing (Ventner projects)
  2. Amazon Rain Forest Microbial Diversity Observatory, see [blog post](http://ivory.idyll.org/blog/jgr-assembling-the-amazon.html) (ARMO data set; Jim Tiedje)
  3. Grand Prairie Challenge Data (Jim Tiedje)

Use of median k-mer coverage (key aspect of comparing "individuals" between samples)

QP is able to use IGS to measure alpha and beta diversity between samples without using BLAST or assembly.

#### Josh's Questions

1. How does IGS compare to OTUs from 16S -- do the techniques compare directly?
2. We want to know taxonomy and function -- can you use the information from IGS to bin sequence reads to better assemble metagenomes to get taxonomy and function?
3. strain level assessment -- how does this k-mer approach deal with microbial strains?
4. Compute IGS before or after diginorm?

____________________________________

Estimated 50 terabytes of unique data Jason Gams 2005

4 petabase pairs of DNA Gilbert 2013

Nature reviews genetics 805-814

Chao estimator

dbg-graph-null for generating random reads
